Libplanet/Store/LiteDBStore.cs

```diff
@@ -0,0 +1,631 @@
+        {
+            long branchPointIndex = branchPoint.Index;
+            List<LiteFileInfo> files =
+                _db.FileStorage
+                    .Find($"stateref/{sourceNamespace}")
+                    .ToList();
+
+            if (!files.Any())
+            {
+                throw new NamespaceNotFoundException(
+                    sourceNamespace,
+                    "The source namespace to be forked does not exist.");
+            }
+
+            foreach (LiteFileInfo srcFile in files)
+            {
+                string destId =
+                    $"stateref/{destinationNamespace}/{srcFile.Filename}";
+                _db.FileStorage.Upload(
+                    destId,
+                    srcFile.Filename,
+                    new MemoryStream());
+
+                LiteFileInfo destFile = _db.FileStorage.FindById(destId);
+                using (LiteFileStream srcStream = srcFile.OpenRead())
+                using (LiteFileStream destStream = destFile.OpenWrite())
+                {
+                    while (srcStream.CanRead)
+                    {
+                        var hashBytes = new byte[HashDigest<SHA256>.Size];
+                        var indexBytes = new byte[sizeof(long)];
+
+                        srcStream.Read(hashBytes, 0, hashBytes.Length);
+                        srcStream.Read(indexBytes, 0, indexBytes.Length);
+
+                        destStream.Write(hashBytes, 0, hashBytes.Length);
+                        destStream.Write(indexBytes, 0, indexBytes.Length);
+
+                        long currentIndex =
+                            BitConverter.ToInt64(indexBytes, 0);
+
+                        if (branchPointIndex <= currentIndex)
+                        {
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <inheritdoc/>
+        public long GetTxNonce(string @namespace, Address address)
+        {
+            var fileId = $"nonce/{@namespace}/{address.ToHex()}";
+            LiteFileInfo file = _db.FileStorage.FindById(fileId);
+
+            if (file is null)
+            {
+                return 0;
+            }
+
+            int hashSize = HashDigest<SHA256>.Size;
+            int blockIndexSize = sizeof(long);
+            int nonceSize = sizeof(long);
+            int nonceEntrySize = hashSize + blockIndexSize + nonceSize;
+
+            using (LiteFileStream stream = file.OpenRead())
+            {
+                if (stream.Length % nonceEntrySize != 0)
+                {
+                    throw new FileLoadException(
+                        $"Nonce file size {stream.Length} should be " +
+                        $"a multiple of nonce entry size {nonceEntrySize}");
+                }
+
+                var buffer = new byte[nonceEntrySize];
+                var offset = (int)(stream.Length - buffer.Length);
+                stream.Read(buffer, offset, buffer.Length);
```

Comment:
The offset should be 0 because the buffer length and the size to read are the same.
I think we should move the file pointer before we read the file.

Reply:
I think moving and reading the file pointer and specifying the `offset` to read(=the last entry's location) are the same behavior. Is it different?

Reply:
According to the [documentation](https://docs.microsoft.com/en-us/dotnet/api/system.io.stream.read?view=netframework-4.8#System_IO_Stream_Read_System_Byte___System_Int32_System_Int32_), the offset parameter is the starting point of the buffer to be copied, not the file stream position.